### PR TITLE
add AWS lambda pubsub example

### PR DESCRIPTION
**Pull request by yuvalif**
see README for details on how to build and run.
note that an AWS account with lambda enabled is required

### CODE REVIEW

**Comment by rootfs** [comment]
CI failed
```
# github.com/ceph/rgw-pubsub-api/go/examples/aws-lambda-source/receiver
go/examples/aws-lambda-source/receiver/main.go:34:53: undefined: s3AccessID
go/examples/aws-lambda-source/receiver/main.go:34:65: undefined: s3AccessKey
go/examples/aws-lambda-source/receiver/main.go:34:78: undefined: s3Endpoint
go/examples/aws-lambda-source/receiver/main.go:48:47: undefined: googleVisionAPIKey
```

The aws vendor pkg should go to https://github.com/ceph/rgw-pubsub-api/tree/master/go/vendor/github.com

**Comment by yuvalif** [comment]
the travis script does not call "go generate ./..."
this is now required (o/w you have to hardcode credential in the lambda function code)

File: go/examples/aws-lambda-source/receiver/main.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/ceph/rgw-pubsub-api/go/examples/knative-eventing-source/vision/pkg/googlevision"
+	"github.com/ceph/rgw-pubsub-api/go/pkg"
+	"log"
+)
+
+type MyResponse struct {
+	Message string `json:"answer"`
+}
+
+//go:generate ./secret_vars.sh
```
Comment (rootfs) [inline go/examples/aws-lambda-source/receiver/main.go]: why not using env var?

File: go/examples/aws-lambda-source/receiver/secret_vars.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+echo "// Code generated by $0; DO NOT EDIT." > secret_vars.go
+echo "package main" >> secret_vars.go
+echo "" >> secret_vars.go
+echo "var (" >> secret_vars.go
+echo -e "\tgoogleVisionAPIKey = \"$GOOGLE_VISION_API_KEY\"" >> secret_vars.go
```
Comment (rootfs) [inline go/examples/aws-lambda-source/receiver/secret_vars.sh]: why code them at compile time?

File: go/examples/aws-lambda-source/test-receiver/main.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+type MyEvent struct {
+	Name string `json:name`
+	Age  int    `json:age`
+}
+
+type MyResponse struct {
+	Message string `json:"Answer:"`
```
Comment (rootfs) [inline go/examples/aws-lambda-source/test-receiver/main.go]: variable name should be the same as json key

File: go/examples/aws-lambda-source/receiver/secret_vars.sh (hunk 2 above)
Comment (yuvalif) [reply go/examples/aws-lambda-source/receiver/secret_vars.sh]: i cannot use env variables in lambda, so they have to be set at compile time.
to make sure they credentials are not commited into github, i use code generation

File: go/examples/aws-lambda-source/test-receiver/main.go (hunk 3 above)
Comment (yuvalif) [reply go/examples/aws-lambda-source/test-receiver/main.go]: this is copy&paste from AWS sample code, but i can change it...

File: go/examples/aws-lambda-source/receiver/secret_vars.sh (hunk 2 above)
Comment (rootfs) [reply go/examples/aws-lambda-source/receiver/secret_vars.sh]: does this doc help? https://docs.aws.amazon.com/lambda/latest/dg/env_variables.html

Comment (yuvalif) [reply go/examples/aws-lambda-source/receiver/secret_vars.sh]: thanks!  will fix

File: go/examples/aws-lambda-source/test-receiver/main.go (hunk 3 above)
Comment (yuvalif) [reply go/examples/aws-lambda-source/test-receiver/main.go]: removed the test code

File: go/examples/aws-lambda-source/receiver/main.go (hunk 1 above)
Comment (yuvalif) [reply go/examples/aws-lambda-source/receiver/main.go]: using env vars now

**Pull request merged**